context/app/static/js/pages/Vis/Vis.jsx

```diff
@@ -33,55 +33,94 @@ import useSearchData from 'js/hooks/useSearchData';
 
  */
 
-const donorRaceQuery = {
+const donorRaceSexQuery = {
   size: 0,
-  query: {
-    bool: {
-      filter: {
-        term: {
-          entity_type: 'donor',
-        },
-      },
-    },
-  },
   aggs: {
-    'mapped_metadata.race': {
-      terms: {
-        field: 'mapped_metadata.race.keyword',
+    composite_data: {
+      composite: {
+        sources: [
+          {
+            'mapped_metadata.race': {
+              terms: {
+                field: 'mapped_metadata.race.keyword',
+              },
+            },
+          },
+          {
+            'mapped_metadata.blood_type': {
+              terms: {
+                field: 'mapped_metadata.blood_type.keyword',
+              },
+            },
+          },
+        ],
+        size: 10000,
       },
     },
   },
 };
 
 function Vis() {
   const { elasticsearchEndpoint, nexusToken } = useContext(AppContext);
-  const { searchData } = useSearchData(donorRaceQuery, elasticsearchEndpoint, nexusToken);
+
+  const { searchData } = useSearchData(donorRaceSexQuery, elasticsearchEndpoint, nexusToken);
   if (!('aggregations' in searchData)) {
     return null;
   }
-  const { buckets } = searchData.aggregations['mapped_metadata.race'];
+
+  function getCount(buckets, bloodType, race) {
+    const filtered = buckets.filter(
+      (b) => b.key['mapped_metadata.blood_type'] === bloodType && b.key['mapped_metadata.race'] === race,
+    );
+    return filtered.length ? filtered[0].doc_count : 0;
+  }
+  function getKeyValues(buckets, key) {
+    return [...new Set(buckets.map((b) => b.key[key]))];
+  }
+  const { buckets } = searchData?.aggregations.composite_data;
 
   return (
-    <Paper>
-      <StyledTableContainer>
-        <Table stickyHeader>
-          <TableHead>
-            <TableRow>
-              {buckets.map((bucket) => (
-                <HeaderCell key={bucket.key}>{bucket.key}</HeaderCell>
-              ))}
-            </TableRow>
-          </TableHead>
-          <TableBody>
-            <TableRow>
-              {buckets.map((bucket) => (
-                <TableCell>{bucket.doc_count}</TableCell>
-              ))}
-            </TableRow>
-          </TableBody>
-        </Table>
-      </StyledTableContainer>
-    </Paper>
+    Object.keys(searchData).length && (
+      <Paper>
+        <StyledTableContainer>
+          <Table stickyHeader>
+            <TableHead>
+              <TableRow>
+                <HeaderCell> </HeaderCell>
+                {getKeyValues(buckets, 'mapped_metadata.blood_type').map((type) => (
+                  <HeaderCell> {type} </HeaderCell>
+                ))}
```

Comment:
This is good, but you can't be certain the order coming out of this function will be the same as what you list below. I'd suggest taking this in stages:
- In this PR, replace this `map` with an explicit list.
- After it's merged, come back and add `const bloodTypes = getKeyValues(buckets, 'mapped_metadata.blood_type')` at the top, and use it here, and in the rows below.
- and finally wrap the rows on the table in a similar loop for race.